### PR TITLE
Jk/feature/local storage

### PR DESCRIPTION
**Pull request by JeffKersting**
#### Purpose:
 > Check all that apply.
- [x] Feature
- [x] Refactor
- [ ] Bug Fix 
- [ ] Clean Up

### What does this PR do?
 > Clearly and concisely describe changes made. 

Adds the loginUser query to Login component. Sets the retrieved user ID to local storage. Passes ID to User-Landing. 
Update User-Landing to incorporate user ID from local storage, and passes that to Update-Form.

Issues closed #

Issue with passing `isLoggedIn` property to homepage properly.

> Describe your code changes in detail for reviewers.

Mostly new code written for setting/retrieving local storage ID value. Some refactoring to query invocation on User-Landing and Update-Form to use the ID from local storage.

App, Login, User-Landing, Update-Form.

N/A

### CODE REVIEW

File: src/GraphQL/signin-user.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { gql } from "@apollo/client";
+
+export default gql`
+  mutation ($email: String!, $password: String!) {
```
Comment (gaj23) [inline src/GraphQL/signin-user.js:4]: Maybe for our testing we need to pay more attention to this?

File: src/GraphQL/update-user.js [hunk 2]
```diff
@@ -1,57 +1,22 @@
-
-import { gql } from '@apollo/client'
-// import GET_USER from './queries'
+import { gql } from '@apollo/client';
 
 export default gql `
-  mutation ($id: ID!, $description: String!) {
+  mutation ($id: ID!, $description: String!, $street: String!, $city: String!, $country: String!, $state: String!, $zip: String!) {
```
Comment (gaj23) [inline src/GraphQL/update-user.js:4]: Great to have this addition!

File: src/components/App/App.js [hunk 3]
```diff
@@ -16,7 +16,18 @@ import Error from '../Error/Error';
 import './_App.scss';
 
 const App = () => {
-  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(null);
+  const [userID, setID] = useState(null)
+
+  useEffect(() => {
+    const storedData = JSON.parse(localStorage.getItem("userData"))
```
Comment (gaj23) [inline src/components/App/App.js:23]: Who knew localStorage would save the day in mod4!

File: src/components/User-Landing/User-Landing.js [hunk 4]
```diff
@@ -10,67 +11,91 @@ import { client } from '../../index'
...
+
+  const [updating, setUpdating] = useState(false);
+  const { loading } = useQuery(GET_USER, {
+    variables: { id: id },
+    onError: <Redirect to="/Error" />
+  });
   const queryData = client.readQuery({
     query: GET_USER,
-    variables: { id: 3,}
+    variables: { id: id },
```
Comment (gaj23) [inline src/components/User-Landing/User-Landing.js:23]: Awesome to have this be dynamic!

**Review by gaj23 — COMMENTED** [review]
Great to have. We should merge this & then where we're doing testing should be merge, too. That way it's up to date. 

I think we'll need an issue to address the fact that the Home page isn't getting the memo that a user is logged in after refresh.

**Review by gaj23 — APPROVED** [review]

**Review by kelsiebesingeryeh — APPROVED** [review]

**Pull request merged**